### ISSUE TITLE
fill_cache kernel writes padding rows when fill_len is not tile-aligned

### ISSUE THREAD

**Issue by sdjordjevicTT** [tenstorrent/tt-metal#42779]
### Summary

The fill_cache multi-core program factory computes the number of tiles to write from `input_tensor.padded_shape()`, not the logical shape. When `fill_len % 32 != 0`, TTNN pads the input tensor up to the next tile boundary. The kernel then writes the full padded tile — including garbage/zero padding rows — into the cache, overwriting valid data.

### Root Cause
In fill_cache_multi_core_program_factory.cpp:34:
```
uint32_t num_blocks_of_work = input_tensor.padded_shape()[1] * input_tensor.padded_shape()[-2] / TILE_HEIGHT;
```
This uses the padded height. For `fill_len=28`, the padded shape is 32, giving 1 tile-row of work. The writer kernel (writer_unary_interleaved_start_id.cpp) writes `num_blocks_per_core * Wt` tiles sequentially — it has no concept of valid vs padding rows within a tile. The 4 padding rows (rows 28–31) contain zeros or garbage and overwrite the corresponding cache positions.

### Repro

Run the following TTNN test to reproduce the issue:
```Python
import ttnn
import torch

def compute_pcc(a, b):
    a_flat = a.flatten().float()
    b_flat = b.flatten().float()
    return torch.corrcoef(torch.stack([a_flat, b_flat]))[0, 1].item()

device = ttnn.open_device(device_id=0)

torch.manual_seed(42)
cache_torch = torch.randn(1, 1, 32, 32, dtype=torch.float32)
input_torch = torch.randn(1, 1, 28, 32, dtype=torch.float32)

# Golden: cache[:, :, :28, :] = input, rows 28-31 remain original cache
golden = cache_torch.clone()
golden[:, :, :28, :] = input_torch

    # Run on device
cache_ttnn = ttnn.from_torch(cache_torch, layout=ttnn.Layout.TILE, device=device)
input_ttnn = ttnn.from_torch(input_torch, layout=ttnn.Layout.TILE, device=device)

ttnn.fill_cache(cache_ttnn, input_ttnn, 0)
result_torch = ttnn.to_torch(cache_ttnn)

# Compare
print("=== Bug 1: partial tile corruption (fill_len=28, non-tile-aligned) ===")
print(f"Cache shape: {cache_torch.shape}, Input shape: {input_torch.shape}")
print()

# Check filled region (rows 0-27)
filled_match = torch.allclose(result_torch[:, :, :28, :], golden[:, :, :28, :])
print(f"Filled region (rows 0-27) matches golden:    {filled_match}")

# Check preserved region (rows 28-31) — this is where the bug hits
preserved_result = result_torch[:, :, 28:32, :]
preserved_golden = golden[:, :, 28:32, :]
preserved_match = torch.allclose(preserved_result, preserved_golden)
print(f"Preserved region (rows 28-31) matches golden: {preserved_match}")
print(f"  Expected (rows 28-31, first 4 vals): {preserved_golden.flatten()[:4].tolist()}")
print(f"  Got      (rows 28-31, first 4 vals): {preserved_result.flatten()[:4].tolist()}")

pcc = compute_pcc(result_torch, golden)
overall_match = torch.allclose(result_torch, golden)
print(f"\nPCC: {pcc:.6f}, allclose: {overall_match}")
passed = overall_match
print(f"PASS: {passed}")
print()

ttnn.close_device(device)
```

### Expected behavior

`fill_cache(cache, input)` should be equivalent to:
```
result = cache.clone()
result[:, :, :fill_len, :] = input
```
Rows fill_len : cache_len must remain unchanged regardless of tile alignment.

**Comment by sdjordjevicTT** [comment]
Hi @cmaryanTT, I am not sure who the owner of kv_cache ops is. Can you please involve the right folks here?

**Comment by sdjordjevicTT** [comment]
Hi @uaydonat, is there a chance this can be addressed soon?
